content/what-is-cross-origin-resource-sharing/index.md

```diff
@@ -0,0 +1,166 @@
+
+More concretely, CORS is a way for web servers to say "Accept cross-origin requests from this origin" or "Do not accept cross-origin requests from this origin".
+
+This is important because cross-origin requests can be quite scary. I could be logged into my bank account and on visiting a malicious site, it could make requests to the bank's servers without my knowledge, and if CORS rules didn't exist, the request would go through - potentially changing or leaking my account information.
+
+CORS is a protocol that defines the limitations of cross-origin requests. These limitations are enforced by our browsers. As a result, we can still make cross-origin requests while still maintaining a high level of security. By specifying which origins are allowed to make requests and which methods and headers are allowed, the browser makes sure that malicious actors can't retrieve sensitive data with cross-origin requests.
+
+## CORS History
+
+CORS was invented to extend and add flexibility to the Same Origin Policy (SOP).
+
+Same origin policy is essentially what the name suggests - resources can only be loaded from the same origin. Two origins are defined as the same if the protocol, port (if specified), and host are the same.
+
+From a technical perspective, an origin can still request a resource from another origin, but the browser prevents the response from being readable.
+
+However, sometimes, we still need to access resources from other origins - such as from `fonts.com`. That's where CORS comes in. CORS relaxes the Same Origin Policy by defining trusted or allowed origins, methods, and headers.
+
+## CORS preflight request
+
+So now, let's get into the actual motion of what happens when requesting resources from another domain.
+
+We've set up an example website at `emailpassword.demo.supertokens.com` where we can see the full CORS motion. This website calls an API on `api-emailpassword.demo.supertokens.com`. Even though both domains are subdomains of `demo.supertokens.com`, our browsers register them as different origins. And thus, CORS comes into the picture. If we had two websites like `example.com` and `fonts.com`, our browser will also register them as different origins. 
+
+During sign in, if you open the browser's dev tools and see the network tab, you will see the preflight request being made. More specifically, the preflight request is an `OPTIONS` request made to our API domain with a couple of headers. Let's take a look at what happens when we click sign in -
+
+```
+OPTIONS /auth/user/email/verify HTTP/1.1
+Host: http://api-emailpassword.demo.supertokens.com
+Access-Control-Request-Method: GET
+Access-Control-Request-Headers: content-type,fdi-version,rid
+Origin: https://emailpassword.demo.supertokens.com
+```
```

Comment:
This isn't the sign in request. It's the email verification request. It's confusing.. change this with the sign in request information.